.github/workflows/snapshot-release.yml

```diff
@@ -50,4 +52,121 @@ jobs:
             Version: `${{ steps.set-snapshot-version.outputs.VERSION }}`
             Repository: `https://s01.oss.sonatype.org/content/repositories/snapshots/`
           edit-mode: replace
-          token: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+  # This job depends on the publish job to finish so we can use the snapshot version to verify
+  # gradle compatibility of the sample apps with actual releases of the SDK with latest changes.
+  gradle-compatibility-builds:
+    name: ${{ matrix.sample-app }} build with AGP ${{ matrix.agp-version }}
+    needs: publish
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # Matrix of sample apps and AGP versions to test
+        # List all sample apps here that you want to test with the AGP versions.
+        sample-app-with-agp:
+          - "java_layout_current"
+          - "kotlin_compose_current"
+          - "java_layout_7"
+          - "kotlin_compose_7"
+        include:
```

Comment:
I couldn't think to better design this matrix, but for the combination we needed, I think this suits best.